src/System.Memory/src/System/Buffers/SequenceReaderExtensions_binary.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    public static partial class SequenceReaderExtensions
+    {
+        /// <summary>
+        /// Try to read the given type out of the buffer if possible. Warning: this is dangerous to use with arbitrary
+        /// structs- see remarks for full details.
+        /// </summary>
+        /// <remarks>
+        /// IMPORTANT: The read is a straight copy of bits. If a struct depends on specific state of it's members to
+        /// behave correctly this can lead to exceptions, etc. If reading endian specific integers, use the explicit
+        /// overloads such as <see cref="TryReadLittleEndian(ref SequenceReader{byte}, out short)"/>
+        /// </remarks>
+        /// <returns>
+        /// True if successful. <paramref name="value"/> will be default if failed (due to lack of space).
+        /// </returns>
+        public static unsafe bool TryRead<T>(ref this SequenceReader<byte> reader, out T value) where T : unmanaged
```

Comment:
Is the `unsafe` keyword necessary here? The T is constrained to unmanaged. Can this be reasoned about by the compiler and allowed without unsafe? Maybe we can use `Unsafe.SizeOf<T>` instead?